### ISSUE TITLE
Monitor for incorrect validators behavior

### ISSUE THREAD

**Issue by akolotov** [poanetwork/bridge-monitor#16]
The monitor functionality must be extended to be able to identify the situation when the validators were compromised and they are trying to perform incorrect actions.
The suggestion is to raise an alarm if:
1. `executeSignatures` is called on the Foreign side but there is no corresponding initiating transaction on the Home side or the value of the transaction differs from the value used in input data of `executeSignatures`
2. `executeAffirmation` is called on the Home side but there is no corresponding intiating transaction on the Foreign side or the value of the transaction differs from the value used in input data of `executeAffirmation`.

**Comment by fernandomg** [comment]
@akolotov, as there's no _alarm_ system implemented in the current monitor version. It's proposed as one of the rehauling steps (check: https://github.com/poanetwork/bridge-monitor/issues/14).

1. Is it ok to create a new endpoint (say `/alerts`) to expose this information or you prefer to add it to one of the existing ones?
1. This functionality should work for all the bridge modes?

If there's any further concern, please let me know.

**Comment by akolotov** [comment]
@fernandomg what endpoints we have now and what they intend to be used for?
Yes, my suggestion is to add new endpoint or extend an existing one to report about an issue in form of: 
```json
{
    "executeSignatures": {
	    "misbehavior": {
	    	"last60blocks": true,
	    	"since61to180blocks": false,
	    	"since181to720blocks": true,
	    	"since721to17280blocks": false,
	    	"erlier17280blocks": false
	    },
	    "mostRecentTxHash": "0x39d5350005310f0e85959368f19756fe892fe21c1eb4eb3ff46fcd39dd68e35f",
        "transactions": {
            "0x39d5350005310f0e85959368f19756fe892fe21c1eb4eb3ff46fcd39dd68e35f": {
                "value": "1500000",
                "block": "6758484",
                "referenceTx": "0x7589a8d10d21b19c40fb3ebdaa61223dd9252612c661b7b38115c2e949ea4c4d",
                "validator": "0x3752F0a521b2E383693f658D6FC377953024C207"
            },
            "0xed2172693b6616458628d1fc9e3b18b8495d0f421c7da750df4d52a38bab15f3": {
                "value": "300",
                "block": "6757984",
                "referenceTx": "",
                "validator": "0xbb140FbA6242a1c3887A7823F7750a73101383e3"
            }
        }
    },
    "executeAffirmation": {
	    "misbehavior": {
	    	"last60blocks": false,
	    	"since61to180blocks": false,
	    	"since181to720blocks": false,
	    	"since721to17280blocks": false,
	    	"erlier17280blocks": false
	    },
	    "mostRecentTxHash": "",
        "transactions": {
        }
    },
    "lastChecked": 1539800932
}
```

**Comment by akolotov** [comment]
addressed in #18
